artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/ActiveMQDestination.java

```diff
@@ -464,7 +464,7 @@ public boolean isQueue() {
    // Public --------------------------------------------------------
 
    public String getAddress() {
-      return simpleAddress.toString();
+      return simpleAddress != null ? simpleAddress.toString() : null;
```

Comment:
toString should always return something not null IMO. Can you instead do:

```java
return simpleAddress != null ? simpleAddress.toString() : "<null address>";
```

or simply

```java
   return "" + simpleAddress;
```

Reply:
`toString()` itself *won't* return `null` since it's returning `"ActiveMQQueue[" + getAddress() + "]"`

Reply:
`getAddress()` is returning `null` not `toString()`

Reply:
Theres a static toString method on simple string that handles null from memory surely just use that for cleaness like in other code areas

Reply:
@michaelandrepearce, I don't see any such method on `SimpleString`.

Reply:
E.g. you could do Objects.toString(simpleAddress, null)

Reply:
FWIW, the pattern `.* != null ? .*\.toString\(\) : null` is used in a handful of places throughout the code-base (including elsewhere in `ActiveMQDestination`) whereas `Objects.toString\(.*\)` is used just once.

Reply:
I think we're getting close to a bike shed here.

Reply:
@jbertram when I read your changes here:

```java
   public String getAddress() {
      return simpleAddress != null ? simpleAddress.toString() : null;
```

I thought this was the toString()


my bad!

sorry for the confusion!